mysql/datadog_checks/mysql/metadata.py

```diff
@@ -103,7 +124,26 @@ def _cursor_run(self, cursor, query, params=None):
             raise
 
     def run_job(self):
-        self.report_mysql_metadata()
+        elapsed_time_settings = time.time() - self._last_settings_collection_time
+        if self._settings_enabled and elapsed_time_settings >= self._settings_collection_interval:
+            self._last_settings_collection_time = time.time()
+            try:
+               self.report_mysql_metadata()
+            except Exception as e:
+                self._log.error(
```

Comment:
"You can use self._log.exception which will log the exception context automatically." - nice, will do

"I think the error should be more user-friendly to focus on the user impact." - I relied on the exception content for that i.e `on occured {}".format(e))` So, my expectation is that exception contains a text such as you've described.

Reply:
> So, my expectation is that exception contains a text

`report_mysql_metadata` doesn't catch any exceptions, so the exception that is caught here could be something as abstract as `KeyError: 0`, which would be super confusing to the user (and solutions engineers). It also would not explain what the user impact is for impacted Datadog features.

Reply:
I see. I'd argue that it's a bad exception then.

I think I made a suboptimal decision. Now, when I think about it DBMAsync has a bit of logic when catching exceptions that I didn't port. My idea was to allow schema collections run even if `report_mysql_metadata()` throws. But I feel like its better to allow everything to fail but be backward compatible with this - 
[https://github.com/DataDog/integrations-core/blob/b93c6fd18bf06a368096e095eba0da37e7a1c644/datadog_checks_base/datadog_checks/base/utils/db/utils.py#L332](url)

Reply:
> I'd argue that it's a bad exception then.

An exception can happen anywhere, including third party packages. And just because an exception is meaningful to developers, does not it will be meaningful to our customers. We don't control all of the exceptions that get thrown up to this level, unless we explicitly catch them and throw more meaningful ones (which is not the case for this code block).

> But I feel like its better to allow everything to fail but be backward compatible with this - https://github.com/DataDog/integrations-core/blob/b93c6fd18bf06a368096e095eba0da37e7a1c644/datadog_checks_base/datadog_checks/base/utils/db/utils.py#L332

Wait that seems worse. If settings fails, so would schemas. What you have implemented now is good in terms of which exceptions are caught. It is an outer wrapper for each separate feature, which ensures they run independently. But the error message will go to the customer's logs. Thus it should be an error that makes sense to the user and indicates which feature they are losing. As it is written now `While executing report_mysql_metadata, the following exception occured` the user would not even know is was collecting settings that failed.

Reply:
"Wait that seems worse. If settings fails, so would schemas. " - 
yes, schemas would fail if setting fail that's the drawback. But you know by putting 2 jobs in a class that is designed to accommodate for one we've already made a decision. So, for me it's just the consequence of this design decision.

I don't understand what you mean by "As it is written now `While executing report_mysql_metadata, the following exception occured`" may be we do a small 5 min sync today? The best if we take a look together at the DBMAsync code and make a quick decision.